rfd/0208-proxy-desktop-session.md

```diff
@@ -0,0 +1,112 @@
+
+## Details
+
+In the Web UI, the TDP stream is transmitted over a cookie-authenticated WebSocket. 
+
+However, Teleport Connect (which uses tsh under the hood) does not rely on cookies for authentication—instead, it uses client certificates.
+To support this scenario, a new gRPC-based API called `ProxyDesktopSession` will be added to the `TransportService` in the Teleport Proxy. 
+
+### Architecture
+
+```
+            gRPC bidi stream   
++-----+   (ProxyDesktopSession)    +----------------+      TCP connection     +-------------------------+
+| tsh | <=======================>  |  Proxy Service | <=====================> | Windows Desktop Service |
++-----+                            +----------------+                         +-------------------------+
+        <===================================================================>
+                              TLS connection (end-to-end)
+```
+
+`ProxyDesktopSession` won't transmit TDP messages directly.
+Instead, it will tunnel a TLS connection over gRPC, which will then carry the TDP messages.
+
+This design simplifies per-session MFA by allowing the client to present an MFA-verified certificate over a single TLS connection that terminates at the Windows Desktop Service.
+
+### Session Initialization Flow
+
+1. The connection is initiated by tsh, which opens the `ProxyDesktopSession` gRPC stream.
+   * The first message sent by tsh must contain the `dial_target`, specifying the desktop name and target cluster. 
+   * If the first message does not include the `dial_target`, the Proxy will immediately terminate the connection.
+2. The Teleport Proxy uses the provided `dial_target` to dial the Desktop Service, following the logic in a Web UI handler.
+   * The Proxy queries the cluster's desktops looking for the ones that match the `desktop_name` in `dial_target`.
+   * The returned desktops are then mapped to desktop services reporting them.
+   * The Proxy attempts to connect to these services in a randomized order until one succeeds.
+3. Data tunneling starts.
+   * Once a Desktop Service is successfully reached, the Proxy begins forwarding raw data between tsh and the Desktop Service.
+   * Meanwhile, tsh initiates a TLS connection over the gRPC stream using its MFA-verified client certificate.
+   * The TLS connection is established when the Desktop Service accepts the handshake.
+
+Note that Proxy won't enforce RBAC checks. It will attempt to connect to the Windows Desktop Service even if the user doesn’t have access to the target desktop.
+The Desktop Service will enforce access control and return an error if the user isn’t allowed.
+
+### Drawbacks and considered alternatives
+
+Tunneling a TLS connection over a gRPC stream introduces some overhead due to the added TLS metadata in each message. 
+This could impact performance slightly, although manual testing did not show any noticeable latency.
```

Comment:
Something has to run the TLS connection to the service tho, and my immediate thought is that it's better to spread the cost of that to the clients rather than have the proxy do it for everyone.